docs/website/docs/plus/ecosystem/delta.md

```diff
@@ -1,33 +1,191 @@
 ```
+</TabItem>
+</Tabs>
+
+Or on the `pipeline.run` level: <!-- can this also be defined in the yaml??-->
+
+```py
+pipeline.run(write_disposition={"disposition": "merge", "strategy": "upsert"})
+```
+
+## Partitioning
+
+Delta tables can be partitioned (using [Hive-style partitioning](https://delta.io/blog/pros-cons-hive-style-partionining/)) by specifying one or more partition column hints on the source/resource level:
+
+<Tabs values={[{"label": "dlt.yml", "value": "yaml"}, {"label": "Python", "value": "python"}]}  groupId="language" defaultValue="yaml">
+  <TabItem value="yaml">
+
+  ```yaml
+  sources:
+    my_source:
+      type: sources.my_source
+      with_args:
+        columns:
+          foo:
+            partition: True
+  ```
+
+  </TabItem>
+  <TabItem value="python">
+
+  ```py
+  @dlt.resource(
+    columns={"foo": {"partition": True}}
+  )
+  def my_resource():
+      ...
+
+  pipeline = dlt.pipeline("loads_delta", destination="delta")
+  ```
+
+  </TabItem>
+</Tabs>
+
+:::caution
+Partition evolution (changing partition columns after a table has been created) is not supported.
+:::
+
+## Table access helper functions
+You can use the `get_delta_tables` helper functions to access the native [DeltaTable](https://delta-io.github.io/delta-rs/api/delta_table/) objects. 
+
+```py
+from dlt.common.libs.deltalake import get_delta_tables
+
+...
+
+# get dictionary of DeltaTable objects
+delta_tables = get_delta_tables(pipeline)
+
+# execute operations on DeltaTable objects
+delta_tables["my_delta_table"].optimize.compact()
+delta_tables["another_delta_table"].optimize.z_order(["col_a", "col_b"])
+# delta_tables["my_delta_table"].vacuum()
+# etc.
+```
+
+## Table format
+The Delta destination automatically assigns the `delta` table format to all resources that it will load. You can still fall back to storing files  by setting `table_format` to native on the resource level:
+
+  ```py
+  @dlt.resource(
+    table_format="native"
+  )
+  def my_resource():
+      ...
+
+  pipeline = dlt.pipeline("loads_delta", destination="delta")
+  ```
+
+### Storage options
+You can pass storage options by configuring `destination.delta.deltalake_storage_options`:
+
+```toml
+[destination.delta]
+deltalake_storage_options = '{"AWS_S3_LOCKING_PROVIDER": "dynamodb", "DELTA_DYNAMO_TABLE_NAME": "custom_table_name"}'
+```
+
+`dlt` passes these options to the `storage_options` argument of the `write_deltalake` method in the `deltalake` library. Look at their [documentation](https://delta-io.github.io/delta-rs/api/delta_writer/#deltalake.write_deltalake) to see which options can be used.
+
+You don't need to specify credentials here. `dlt` merges the required credentials with the options you provided before passing it as `storage_options`.
+
+>❗When using `s3`, you need to specify storage options to [configure](https://delta-io.github.io/delta-rs/usage/writing/writing-to-s3-with-locking-provider/) locking behavior.
```

Comment:
```suggestion
:::warning
When using `s3`, you need to specify storage options to [configure](https://delta-io.github.io/delta-rs/usage/writing/writing-to-s3-with-locking-provider/) locking behavior.
:::
```
Which storage options should I specify for s3? Not clear, should it be `'{"AWS_S3_LOCKING_PROVIDER": "dynamodb", "DELTA_DYNAMO_TABLE_NAME": "custom_table_name"}'` ?